omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -587,11 +581,30 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                     self._table.numberOfPages = Math.floor(initialRowCount/self._table.fnSettings()._iDisplayLength)+1;
 
             } else if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
-                if (self.options.searchPhrase == null) {
-                    self.options.searchPhrase = "";
-                }
-                $j(input).val(self.options.searchPhrase).keyup();
+            	if (self.options.searchPhrase == null) {
+            		self.options.searchPhrase = "";
+            	}                
+            	//check to see if we need to select the checkbox for includeVerbose and/or includeVoided
+            	//since there is a searchPhrase the page will reload using these values
+            	if(self.options.showVerbose){
+            		$j('#includeVerbose').attr('checked','checked');
+            	}
+            	if(self.options.includeVoided){
+            		$j('#includeVoided').attr('checked','checked');
+            	}           	
             }
+            $j(input).val(self.options.searchPhrase).keyup();
+        	//setting the cursor to the end of the searchPhrase
+        	//setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+        	var inputNodeDOM = $j('#inputNode').get(0);
+        	inputNodeDOM.focus();
+        	if(inputNodeDOM){
+        		var searchPhraseLength = self.options.searchPhrase.length;
+        		//setSelectionRange is for setting the cursor at the end of the search text
```

Comment:
This is actually a common way of putting the cursor at the end of text. It works for Chrome and Firefox on my Ubuntu machine. If you are not satisfied with this solution though I am happy to either not set the cursor at the end or to try something else.